### PR TITLE
Update for python 2 and 3

### PR DESCRIPTION
**Pull request by ashcrow**
This change moves metaclasses, string checks, and unicode casting to six
and also moves from iteritems to items.

**Note**: I found unittests fail with and without this patch.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -12,3 +12,4 @@ PyYAML==3.12
 sh==1.12.13
 tabulate==0.7.7
 testinfra==1.5.5
+six
```
Comment (retr0h) [inline requirements.txt]: Can you keep the list alphabetical and pin to a release of six.

**Comment by retr0h** [comment]
Also, as an FYI the v1 version of molecule is on it's way out in favor of v2.  Might be of interest to look at v2's py3 compatibility.  I believe I did a better job with that one.

**Comment by ashcrow** [comment]
Understood. Closing this PR.

**Comment by retr0h** [comment]
I'm good with merging this, but I wanted to point out v2.

**Comment by ashcrow** [comment]
@retr0h thanks. Unless v1 will live on for quite a while I think taking a gander (and possibly fixing if need be) v2 is probably of more benefit.

**Pull request closed without merging**